aws_s3_file_explorer/tasks/s3_file_explorer.airplane.ts

```
@@ -0,0 +1,175 @@
+import {
+  S3Client,
+  ListObjectsCommand,
+  ListBucketsCommand,
+  GetObjectCommand,
+  PutObjectCommand,
+} from "@aws-sdk/client-s3";
+import {
+  STSClient,
+  AssumeRoleWithWebIdentityCommand,
+} from "@aws-sdk/client-sts";
+import airplane from "airplane";
+
+export const list_s3_buckets = airplane.task(
+  {
+    slug: "list_s3_buckets",
+    parameters: {
+      region: "shorttext",
+    },
+    envVars: {
+      AWS_OIDC_ROLE: { config: "aws_oidc_role_test" },
+    },
+  },
+  async (params) => {
+    const client = await newAWSClient(params.region);
+    const command = new ListBucketsCommand({
+      BucketPolicyOnly: {
+      PolicyOnly: true, 
+      SkipEvaluation: false
+    }});
+    try {
+      const { Owner, Buckets } = await client.send(command);
+      console.log(
+        `Found ${Buckets?.length} bucket${
+          Buckets?.length === 1 ? "" : "s"
+        } in ${Owner?.DisplayName}`
+      );
+      return (
+        Buckets?.map((bucket) => ({
+          name: bucket.Name,
+          creation_date: bucket.CreationDate,
+        })) ?? []
+      );
+    } catch (err) {
+      console.error(err);
```

We probably want to rethrow this error so that we actually get an error if this fails? As written the task will succeed even if the call fails